### ISSUE TITLE
Not receiving notifications after openning the app second time

### ISSUE THREAD

**Issue by Ko09** [aws-amplify/amplify-js#7425]
**Describe the bug**
I sat up notification following this doc https://docs.amplify.aws/lib/push-notifications/getting-started/q/platform/js. I successfully received a notification after launching the app for the first time.But If I open the app second time after killing the app, the app stops receiving notifications. But if I open the app again, the app starts receiving notifications. If I open the app, it stops receiving notifications again, and it continues like this. So somehow my app stops receiving notifications after opening the app second time. I google the issue but I found nothing.  

**To Reproduce**

1. set up notification 
2. update endpoints with userId and the token from AsyncStorage
3. send a notification then check the app receives the notification.
4. kill the app and open it again 
5. after Loading... it stops receiving notifications, but it still receives notifications while it says Loading....

**Expected behavior**
receiving notifications second time open the app

**Code Snippet**
Please provide a code snippet or a link to sample code of the issue you are experiencing to help us reproduce the issue. (Be sure to remove any sensitive data)

```
class App extends React.Component {
  componentDidMount() {
    PushNotification.onNotification((notification) => {
  // Note that the notification object structure is different from Android and IOS
  // required on iOS only (see fetchCompletionHandler docs: https://github.com/react-native-community/push-notification-ios#finish)
  notification.finish(PushNotificationIOS.FetchResult.NoData);
});

// get the registration token
// This will only be triggered when the token is generated or updated.
  PushNotification.onRegister(async(token) => {
    console.log('token', token)
    await AsyncStorage.setItem('deviceToken', token);
});

// get the notification data when notification is opened
PushNotification.onNotificationOpened((notification) => {
    console.log('the notification is opened', notification);
});
  }
  render() {
    return (
       <Home />
    )
  }
};

class Home extends React.Component {
  componentDidMount() {
    updateEndpoint()
  }
  render() {
    return (
       <TabScreen />
    )
  }
};

async function updateEndpoint () {
  const deviceToken = await AsyncStorage.getItem("deviceToken");
  const user = await Auth.currentAuthenticatedUser()
  const result = await Analytics.updateEndpoint({address : deviceToken,userId: user.attributes.sub})
  console.log('result',result)
};

export default withAuthenticator(App,{
  signUpConfig: {
    hiddenDefaults: ["phone_number","email"],
  }
 })

```
**Additional context**
"dependencies": {
    "@aws-amplify/pushnotification": "^3.2.16",
    "@react-native-community/netinfo": "^5.9.9",
    "@react-native-community/push-notification-ios": "^1.8.0",
    "amazon-cognito-identity-js": "^4.5.5",
    "aws-amplify": "^3.3.13",
    "aws-amplify-react-native": "^4.2.10",
    "graphql": "^15.2.0",
    "react": "16.11.0",
    "react-native": "0.62.2",
    "react-native-gesture-handler": "^1.6.1",
    "react-native-reanimated": "^1.9.0",
    "react-native-safe-area-context": "^3.0.7",
    "react-native-screens": "^2.9.0",
    "react-native-vector-icons": "^7.1.0",
    "react-navigation-stack": "^2.8.2",
  },

**Comment by elorzafe** [comment]
@Ko09 

Where are you doing `Amplify.configure`? Is it on the entry point of the App?

**Comment by Ko09** [comment]
I am doing `Amplify.configure` on `index.js` .
I think this file is the entry point of the App by default.
I am using React Native CLI.

```
index.js
import config from './aws-exports'

Amplify.configure(config)
console.log('configured')

const AppWithData = () => (
    <App />
)

AppRegistry.registerComponent(appName, () => AppWithData);
```
using Iphone5s 12.4.9

**Comment by Ko09** [comment]
I upgraded react native but still same result.
I noticed `node_modules/@aws-amplify/pushnotification` uses  "@react-native-community/push-notification-ios": "1.0.3"
Does this mean if I call `pushNotification.onNotification`, am I using "push-notification-ios": "1.0.3" with "react native" :  "0.63.4" ? 

if I am using "push-notification-ios": "1.0.3" , I want to feature request to update `@react-native-community/push-notification-ios ` of `@aws-amplify/pushnotification `to the latest version(1.8.0).
Manually installing nodemodule seems little bit hassle, since it will be overwritten every time I run npm i, but I will try It might worth it.
